packages/app/src/app/components/CreateNewSandbox/CreateSandbox/CreateSandbox.tsx

```diff
@@ -45,28 +51,65 @@ interface CreateSandboxProps {
   isModal?: boolean;
 }
 
+type EssentialsState =
+  | {
+      state: 'loading';
+    }
+  | {
+      state: 'success';
+      essentials: ITemplateInfo[];
+    }
+  | {
+      state: 'error';
+      error: string;
+    };
+
 export const CreateSandbox: React.FC<CreateSandboxProps> = ({
   collectionId,
   initialTab,
   isModal,
 }) => {
-  const { hasLogIn } = useAppState();
+  const { hasLogIn, activeTeamInfo, user } = useAppState();
   const actions = useActions();
 
+  const isRepositoriesPage = location.pathname.includes('/repositories');
+  const isDashboardPage = location.pathname.includes('/dashboard');
+  const isUser = user?.username === activeTeamInfo?.name;
+
+  const [essentialState, setEssentialState] = useState<EssentialsState>({
+    state: 'loading',
+  });
+
   const tab = useTabState({
     orientation: 'vertical',
-    selectedId: initialTab || 'create',
+    selectedId: initialTab || isRepositoriesPage ? 'import' : 'create',
   });
 
   useEffect(() => {
-    /* ❗️ TODO: We can probably take this out of the useEffect and put it into the useTabState selectedId */
-    if (location.pathname.includes('/repositories')) {
-      tab.select('Import');
+    async function getEssentials() {
```

Comment:
nitpick: can you move this to a helper function perhaps? or a custom hook like `useEssentialTemplates`?

Reply:
[Sure can do!](https://github.com/codesandbox/codesandbox-client/pull/6909/commits/f1fc82673c303d08f3d630d8cea857f800823b2d)